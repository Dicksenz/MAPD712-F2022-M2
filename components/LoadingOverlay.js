import React from "react";
import { ActivityIndicator, View, StyleSheet } from "react-native";

const LoadingOverlay = () => {
  return (
    <View style={styles.loading}>
      <ActivityIndicator size="large" />
    </View>
  );
};

const styles = StyleSheet.create({
  loading: {
    backgroundColor: "black",
    height: "100%",
    position: "absolute",
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: "center",
    justifyContent: "center",
    opacity: 0.7,
  },
});

export default LoadingOverlay;
