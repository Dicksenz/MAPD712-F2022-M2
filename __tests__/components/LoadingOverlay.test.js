import React from "react";
import renderer from "react-test-renderer";
import LoadingOverlay from "../../components/LoadingOverlay";

test("Check if LoadingOverlay component renders correctly", () => {
  const tree = renderer.create(<LoadingOverlay />).toJSON();
  expect(tree).toMatchSnapshot();
});
