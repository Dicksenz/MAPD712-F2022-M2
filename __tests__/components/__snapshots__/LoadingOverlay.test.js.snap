// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if LoadingOverlay component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "bottom": 0,
      "height": "100%",
      "justifyContent": "center",
      "left": 0,
      "opacity": 0.7,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <ActivityIndicator
    size="large"
  />
</View>
`;
